homeassistant/components/denonavr/__init__.py

```diff
@@ -32,22 +34,22 @@
 GET_COMMAND_SCHEMA = CALL_SCHEMA.extend({vol.Required(ATTR_COMMAND): cv.string})
 
 SERVICE_TO_METHOD = {
-    SERVICE_GET_COMMAND: {"method": "get_command", "schema": GET_COMMAND_SCHEMA}
+    SERVICE_GET_COMMAND: {"method": "async_get_command", "schema": GET_COMMAND_SCHEMA}
 }
 
 
-def setup(hass: core.HomeAssistant, config: dict):
+async def async_setup(hass: core.HomeAssistant, config: Dict):
     """Set up the denonavr platform."""
 
     def service_handler(service):
         method = SERVICE_TO_METHOD.get(service.service)
         data = service.data.copy()
         data["method"] = method["method"]
-        dispatcher_send(hass, DOMAIN, data)
+        async_dispatcher_send(hass, DOMAIN, data)
 
     for service in SERVICE_TO_METHOD:
         schema = SERVICE_TO_METHOD[service]["schema"]
-        hass.services.register(DOMAIN, service, service_handler, schema=schema)
+        hass.services.async_register(DOMAIN, service, service_handler, schema=schema)
 
```

Comment:
This is reimplementing entity services https://developers.home-assistant.io/docs/dev_101_services/#entity-services.  Delete this block and the dispatcher in `media_player.py`, then add this to `async_setup_entry` in

```
SERVICE_GET_COMMAND = "get_command"
```
```
    platform = entity_platform.current_platform.get()

    platform.async_register_entity_service(
        SERVICE_GET_COMMAND,
        {vol.Required(ATTR_COMMAND): cv.string},
        SERVICE_GET_COMMAND,
    )
```

Reply:
Done